app/assets/javascripts/template_invocation.js

```diff
@@ -24,6 +24,33 @@ function refresh_search_query(value){
   $('textarea#targeting_search_query').val($('span#bookmark_query_map span#bookmark-' + id).data('query'));
 }
 
+function show_preview_hosts_modal() {
+  var modal_window = $('#previewHostsModal');
+
+  var form = $('form#job_invocation_form');
+  var data = form.serializeArray();
+
+  request = $.ajax({
+    data: data,
+    type: 'POST',
```

Comment:
Sorry to jump on this too late, if we send the search field it can be very long (bulk action on 500 hosts, FQDN is ~20 chars), GET search query size is limited by some browsers (IE ~ 2k), Apache refuse after 8k (according to doc) so I think POST would be justified here.